Source/SPUD/Private/SpudSubsystem.cpp

```diff
@@ -716,6 +722,11 @@ void USpudSubsystem::PostLoadStreamLevelGameThread(FName LevelName)
 			UE_LOG(LogSpudSubsystem, Log, TEXT("PostLoadStreamLevel called for %s but level is null; probably unloaded again?"), *LevelName.ToString());
 			return;
 		}
+
+		// Events triggered by loading might need to know that SPUD is loading.
+		auto crtState = CurrentState;
+		CurrentState = ESpudSystemState::LoadingGame;
+
```

Comment:
As with similar previous changes, this is incorrect since it will change the SPUD state to think it's loading a save game when in fact it's just loading a stream level. I think you maybe misunderstood what `CurrentState == ESpudSystemState::LoadingGame` is representing here, it doesn't mean that SPUD is restoring state, it means that a save game is being loaded. It's a top-level state for the entire system, not a flag saying that a level is having its state loaded/saved. Because of streaming, storing / restoring of individual level state can happen independently of the overarching save/load game state.

If there are cases which need to know that a level is having its state restored then we'll need to add something else that tracks this on a per-level basis, but right now I'm not sure what problem this change is trying to fix.

Reply:
Gotcha - but I'm not sure if a flag that tracks if SPUD is restoring a level is going to solve that reliably; it would depend on OnBeginOverlap being triggered during the state restore, during the time when we set some new flag, before resetting it at the end of restoration of all actors, and I'm not sure if that's guaranteed? Couldn't you get the OnBeginOverlap in the frame after, or even in the same frame but after the sequence to restore the level is complete. I think the only way to reliably solve it is to do what you suggest and track whether you've already applied a volume effect or not, perhaps by a (saved) cooldown timer on the volume. You kind of need this anyway to prevent players going back and forth over the threshold of the volume to get the effects continuously.

I think a lot of games avoid having to deal with this this by only doing checkpoints in "safe" areas outside of volumes.

Reply:
Although, regardless, whether a level is currently restoring is still _potentially_ useful information and I don't object to adding something new to the API for that, perhaps in a different PR.

Reply:
I've perhaps simplified too much. A real example: a walk-into pick-up. OnBeginOverlap with the player does its effect, sets a (save-able) flag that it did, turns off collision, plays a disappearance animation, then destroys itself. If the game is saved after stepping on it, but before the actor is destroyed, it is possible that the player gets back to it after loading, even saving while overlapping it.

Overlaps are reported by physics. And physics is guaranteed to have run for 1 frame before SPUD completes restoring state, even if using the level streaming detection with WP.
If the restoring is triggered by loading a saved game, the overlap (assume game saved with the player in the volume) will certainly happen while the "SPUD is restoring state" flag is set.
The only possibility for mishap I see is when using the automatic streamed level detection, if the player is in the overlap volume before SPUD notices that the level has been loaded. But shouldn't really happen, since level are normally loaded before the player enters them.
Being able to detect that SPUD is restoring state is needed even if we have some other state tracking. As I said before, the overlap will happen before SPUD restores state, so we could not rely on checking a saved flag. Even if there might be ways around this, why take the complicated road when all we really need to know if we're in middle of restoring state or not.

I'll create another flag for this.

Reply:
Done. Also fixed the name warning - shouldn't be issued for objects using SpudGuid.